src/soter/openssl/soter_verify_ecdsa.c

```diff
@@ -100,37 +105,35 @@ soter_status_t soter_verify_final_ecdsa_none_pkcs8(soter_sign_ctx_t* ctx,
                                                    const void* signature,
                                                    const size_t signature_length)
 {
-    if (!ctx) {
+    if (!ctx || !ctx->pkey) {
+        return SOTER_INVALID_PARAMETER;
+    }
+    if (!signature || signature_length == 0) {
         return SOTER_INVALID_PARAMETER;
     }
-    EVP_PKEY* pkey = EVP_PKEY_CTX_get0_pkey(ctx->pkey_ctx);
-    if (!pkey) {
+    if (EVP_PKEY_base_id(ctx->pkey) != EVP_PKEY_EC) {
         return SOTER_INVALID_PARAMETER;
     }
 
-    int res = EVP_DigestVerifyFinal(ctx->md_ctx, (unsigned char*)signature, signature_length);
-    switch (res) {
-    case 0:
-        return SOTER_INVALID_SIGNATURE;
-    case 1:
-        return SOTER_SUCCESS;
```

Comment:
ouch

Reply:
IIRC it could have returned `<0` for other failure (invalid params etc), so the idea here was to tell apart invalid input parameters (bug?) vs invalid signature